src/extensions/example-extension/example-extension.tsx

```diff
@@ -0,0 +1,55 @@
+import { Button, DynamicPageType, Icon, LensExtension } from "@lens/extensions"; // fixme: map to generated types from "extension-api.ts"
+import React from "react";
+import path from "path";
+
+let extension: ExampleExtension; // todo: provide instance from context
+
+export default class ExampleExtension extends LensExtension {
+  protected unRegisterPage = Function();
+
+  onActivate() {
+    extension = this
+    console.log('EXAMPLE EXTENSION: ACTIVATE', this.getMeta())
+    const { dynamicPages } = this.runtime;
+
+    this.unRegisterPage = dynamicPages.register({
```

Comment:
Maybe the API should be something like:

```js
this.registerPage({
      type: DynamicPageType.CLUSTER,
      path: "/extension-example",
// ...
})
```
and LensExtension would automatically do unregister etc housekeeping.

Reply:
Done